server/service/transport_osquery.go

```diff
@@ -73,10 +74,25 @@ func decodeSubmitDistributedQueryResultsRequest(ctx context.Context, r *http.Req
 		results[query] = queryResults
 	}
 
+	// Statuses were represented by strings in osquery < 3.0 and now
+	// integers in osquery > 3.0. Massage to string for compatibility with
+	// the service definition.
+	statuses := map[string]string{}
```

Comment:
should we make deeper changes and use `map[string]int` (or float64`) instead of converting the number into a string?

Reply:
I refactored the types (added `kolide.OsqueryStatus`). This may be worth reviewing again.